lib/galaxy/schema/fetch_data.py

```diff
@@ -0,0 +1,280 @@
+import json
+from enum import Enum
+from typing import (
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import (
+    BaseModel,
+    Extra,
+    Field,
+    Json,
+    validator,
+)
+from typing_extensions import (
+    Annotated,
+    Literal,
+)
+
+from galaxy.schema.fields import EncodedDatabaseIdField
+from .schema import HistoryIdField
+
+
+class FetchBaseModel(BaseModel):
+    class Config:
+        allow_population_by_field_name = True
+
+
+class ElementsFromType(str, Enum):
+    ARCHIVE = "archive"
+    BAGIT = "bagit"
+    BAGIT_ARCHIVE = "bagit_archive"
+    DIRECTORY = "directory"
+
+
+AutoDecompressField = Field(False, description="Decompress compressed data before sniffing?")
+
+
+class BaseFetchDataTarget(FetchBaseModel):
+    auto_decompress: bool = AutoDecompressField
+
+
+class ItemsFromSrc(str, Enum):
+    url = "url"
+    files = "files"
+    path = "path"
+    ftp_import = "ftp_import"
+    server_dir = "server_id"
```

Comment:
I think this is a typo
```suggestion
    server_dir = "server_dir"
```